packages/roosterjs-editor-plugins/lib/Paste/officeOnlineConverter/convertPastedContentFromWordOnline.ts

```diff
@@ -0,0 +1,227 @@
+//                  <li></li>
+//              </ul>
+//          </div>
+//      </div>
+//      <div class='OutlineGroup'>
+//          <div class="ListItemWrapper">  ----------> list items: for ordered list, each items has it's own wrapper
+//              <ol>
+//                  <li></li>
+//              </ol>
+//          </div>
+//          <div class="ListItemWrapper">
+//              <ol>
+//                  <li></li>
+//              </ol>
+//          </div>
+//      </div>
+//  </body>
+//</html>
+//
+
+/**
+ * Convert text copied from word online into text that's workable with rooster editor
+ * @param doc Document that is being pasted into editor.
+ */
+export default function convertPastedContentFromWordOnline(doc: HTMLDocument) {
+    const listItemBlocks: ListItemBlock[] = getListItemBlocks(doc);
+
+    listItemBlocks.forEach((itemBlock) => {
+
+        flattenListBlock(doc.body, itemBlock);
+
+        itemBlock.insertPositionElement = itemBlock.endElement.nextElementSibling;
+
+        let convertedListElement: Element;
+        itemBlock.listItemContainers.forEach((listItemContainer) => {
+            let listType: string = getContainerListType(listItemContainer); // list type that is contained by iterator.
+
+            // Initialize processed element with propery listType if the
+            if (!convertedListElement) {
+                convertedListElement = document.createElement(listType);
+            }
+
+            // Get all list items(<li>) in the current iterator element.
+            const currentListItems = listItemContainer.querySelectorAll('li');
+            currentListItems.forEach((item) => {
+                insertListItem(convertedListElement, item, listType);
+            })
+        });
+
+        insertConvertedListToDoc(convertedListElement, doc.body, itemBlock);
+
+        const parentContainer = itemBlock.startElement.parentElement;
+        if (parentContainer) {
+            itemBlock.listItemContainers.forEach((listItemContainer) => {
+                parentContainer.removeChild(listItemContainer);
+            });
+        }
+    });
+}
+
+/**
+ * Take all the list items in the document, and group the consecutive list times in a list block;
+ * @param doc pasted document that contains all the list element.
+ */
+function getListItemBlocks(doc: HTMLDocument): ListItemBlock[] {
+    const listElements = doc.getElementsByClassName(LIST_CONTAINER_ELEMENT_CLASS_NAME);
+    const result: ListItemBlock[] = [];
+    let curListItemBlock: ListItemBlock;
+    for (let i = 0; i < listElements.length; i++) {
+        let curItem = listElements[i];
+        if (!curListItemBlock) {
+            curListItemBlock = createListItemBlock(curItem)
+        } else {
+            const { listItemContainers } = curListItemBlock;
+            const lastItemInCurBlock = listItemContainers[listItemContainers.length - 1];
+            if (curItem.isEqualNode(lastItemInCurBlock.nextElementSibling)
```

Comment:
why use isEqualNode? If you want to check if they are the same node, just use ==